### PR TITLE
making TextWriterTraceAdapter synchronized

### PR DESCRIPTION
**Pull request by brettsam**
Addresses #1042

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TextWriterTraceAdapter.cs [hunk 1]
```diff
@@ -27,14 +27,18 @@ public override Encoding Encoding
...
+        }
 
         public override void Write(char value)
         {
             // buffer all output
             _text.Append(value);
 
             int len = _text.Length;
-            if (len > 2 && _text[len - 2] == '\r' && _text[len - 1] == '\n')
+            if (len >= 2 && _text[len - 2] == '\r' && _text[len - 1] == '\n')
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TextWriterTraceAdapter.cs]: Can you explain this change?

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/Bindings/TraceWriter/TextWriterTraceAdapterTests.cs [hunk 2]
```diff
@@ -70,5 +73,24 @@ public void Flush_FlushesRemainingBuffer()
...
+
+        [Fact]
+        public async Task TestMultipleThreads()
+        {
+            // This validates a bug where writing from multiple threads throws an exception.             
+            TestTraceWriter trace = new TestTraceWriter(TraceLevel.Verbose);
+            TextWriter adapter = TextWriterTraceAdapter.Synchronized(trace);
+
+            // Start Tasks to write
+            List<Task> tasks = new List<Task>();
```
Comment (mathewc) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/Bindings/TraceWriter/TextWriterTraceAdapterTests.cs]: Did this test repro the problem reliably?

File: src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TextWriterTraceAdapter.cs (hunk 1 above)
Comment (brettsam) [inline src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TextWriterTraceAdapter.cs]: In my test, I just log WriteLine(string.Empty) -- which looks like it just sends '\r\n'. In that case, the string length was only 2, so this logic didn't kick in like I would have expected. Instead, it waited until it had '\r\n\r\n' before flushing. It's a minor change -- but I thought it made sense.

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/Bindings/TraceWriter/TextWriterTraceAdapterTests.cs (hunk 2 above)
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/Bindings/TraceWriter/TextWriterTraceAdapterTests.cs]: Yeah, every time.

**Pull request merged**